### PR TITLE
tuba: 0.7.2 -> 0.8.1

### PR DESCRIPTION
**Pull request by Aleksanaa**
## Description of changes

Added clapper support

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by chuangzhu — APPROVED** [review]
Result of `nixpkgs-review pr 321736` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tuba</li>
  </ul>
</details>

File: pkgs/applications/misc/tuba/default.nix [hunk 1]
```diff
@@ -67,20 +70,26 @@ stdenv.mkDerivation rec {
     gst-plugins-base
     (gst-plugins-good.override { gtkSupport = true; })
     gst-plugins-bad
-  ]);
+  ]) ++ lib.optionals clapperSupport [
```
[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/applications/misc/tuba/default.nix:73]: I may refactor that package, but splitting outputs seems problematic. I'll make it simpler later.

Comment (GeopJr) [inline pkgs/applications/misc/tuba/default.nix:73]: If possible, please add a warning that some stuff might be broken. I worked with Rafostar on this and some bug fixes have not made it into a Clapper release yet

Most critical being this one: https://github.com/GeopJr/Tuba/pull/931#issuecomment-2170028583 which will leave Tuba frozen when closing a video.

**Review by chuangzhu — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/tuba/default.nix (hunk 1 above)
Comment (Aleksanaa) [reply pkgs/applications/misc/tuba/default.nix:73]: Yeah I've seen that

File: pkgs/applications/misc/tuba/default.nix [hunk 2]
```diff
@@ -23,17 +23,20 @@
 , libspelling
 , icu
 , gst_all_1
+, clapper
+, clapperSupport? true
```
[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/applications/misc/tuba/default.nix:27]: Oh sorry, I was testing it, so it's true, but I forgot

**Comment by GeopJr** [comment]
Please wait a bit before merging, I'll make a patch release soon to address some issues!

File: pkgs/applications/misc/tuba/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/applications/misc/tuba/default.nix:75]: I may do that, but splitting output is still tricky by default. I probably won't have much time these days to push this change forward.

**Comment by GeopJr** [comment]
Done https://github.com/GeopJr/Tuba/releases/tag/v0.8.1

**Comment by Aleksanaa** [comment]
Good to go

image

**Review by chuangzhu — APPROVED** [review]
Result of `nixpkgs-review pr 321736` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tuba</li>
  </ul>
</details>

**Pull request merged**